crates/subspace-networking/src/node.rs

```diff
@@ -114,22 +116,48 @@ pub enum SendPiecesByRangeRequestError {
     IncorrectResponseFormat,
 }
 
+#[derive(Debug, Error)]
+pub enum RelayConfigurationError {
+    /// Client configuration error: expected server parent configuration.
+    #[error("cannot configure relay client, parent configuration should be server")]
+    ExpectedServerConfiguration,
+}
+
 /// Implementation of a network node on Subspace Network.
 #[derive(Debug, Clone)]
 pub struct Node {
     shared: Arc<Shared>,
+    relay_config: RelayConfiguration,
 }
 
 impl Node {
-    pub(crate) fn new(shared: Arc<Shared>) -> Self {
-        Self { shared }
+    pub(crate) fn new(shared: Arc<Shared>, relay_config: RelayConfiguration) -> Self {
+        Self {
+            shared,
+            relay_config,
+        }
     }
 
     /// Node's own local ID.
     pub fn id(&self) -> PeerId {
         self.shared.id
     }
 
+    /// Configures relay-client configuration (ClientAcceptor) from this Node. It expects Node
+    /// running in the relay server mode and create relay client's listening address with the proper
+    /// circuit address in the format: /ip4/127.0.0.1/tcp/50000/p2p/<server_peer_id>/p2p-circuit
+    pub fn configure_relay_client(&self) -> Result<RelayConfiguration, RelayConfigurationError> {
+        if let RelayConfiguration::Server(server_addr, _) = self.relay_config.clone() {
+            Ok(RelayConfiguration::ClientAcceptor(
+                server_addr
+                    .with(Protocol::P2p(self.id().into()))
+                    .with(Protocol::P2pCircuit),
+            ))
+        } else {
+            Err(RelayConfigurationError::ExpectedServerConfiguration)
```

Comment:
Yeah, not particularly nice, I think we'll end up with a separate behavior in the end to avoid this.